### DISCUSSION TITLE
Replace the "Search" prompt with say, the time

### DISCUSSION THREAD

**Discussion by stibinator** [davatorium/rofi#2062] [Q&A] [answered]
Is it possible to replace the default "Search" prompt with something dynamically generated like the current time, or even weather? I'm using rofi in i3wm. I tried
```
bindsym $mod+space exec "rofi -modi combi -show combi -combi-modi 'window,drun,run' -p 'ohai'" 
```
to test if I could change the prompt but it still shows the default "Search" text.

I can include the date in the results by doing
```
bindsym $mod+space exec "rofi -modi combi -show combi -combi-modi '>:date,window,drun,run' -p 'ohai'" 
```
but I'd love to have the time and date just appear as the prompt text.

**Comment by DaveDavenport** [comment, marked as answer]
Yes, remove the prompt widget and add a custom textbox widget. See the manpages for more information.

Example:

```bash
rofi -show drun -theme-str 'inputbar { children: [ textbox-myprompt, entry ]; } textbox-myprompt { content: "mytest"; expand: false;  }' 
```

**Reply by stibinator** [reply]
Thanks. If i wanted to do this in the theme file, how would I make the content dynamic?

**Reply by stibinator** [reply]
for example if I wanted to get the weather, like: 
```
curl  "wttr.in/~Coburg+Victoria?format=3"
```

**Reply by DaveDavenport** [reply]
> If i wanted to do this in the theme file, 

you can copy the above in your theme file and pass in the string as an environment variable. See manpage.

**Reply by stibinator** [reply]
Thank you.

**Reply by stibinator** [reply]
I don't seem to be able to use any environment variables in my theme. If I have a script 
```
export WTTR=$(curl  "wttr.in/~Coburg+Victoria?format=3")
rofi -show drun
```
and in my theme I have
```
 *  {
      weather: ${WTTR} ;
or
      weather: env(WTTR, "no weather available");
}
...
textbox-weather {
    content: @weather; 
}
```
I get an error

> Error while parsing theme /home/stib/.config/rofi/config.rasi
> Parser error: syntax error unexpected invalid property value
> Location: line 1 column 1 to line 1 column 23 

The location first column always starts at line 1 col 1, the last column number changes to either 23 or 8. Line 1 of my config.rasi is just:
```
configuration {
``` 
which is 14 chars.
if I set the `$WTTR` environment variable to a number, e.g.
```
export WTTR=1234
rofi -show drun
```
it shows the dialog, but the weather box is empty. If I use the form 
```
*  {
       weather: env(WTTR, "no weather available");
}
```
and unset the `$WTTR` environment variable, 
```
export WTTR=
```
it correctly subs in the default `no weather available` message.

I thought it might be a unicode thing, because of the icons returned by wttr.in, but if I set `$WTTR` to a string, e.g. 
```
export WTTR="FOO"
rofi -show drun
```
Then I get the error message again.
Also if I set `$WTTR` to `FOO` the error reports the syntax error as ending at column 8, if I set it to `FOOBAR` the column number is 23. This smells a bit buggy.

This seems to be a bug, or have I misunderstood how environment variables get used?

**Reply by DaveDavenport** [reply]
Make sure it has the right quoting around it.. so the environment should have the opening/closing quotes (`"`)

**Reply by stibinator** [reply]
Ah, gotcha. Thanks again, it's working now. If anyone is interested, what I've done is add a systemd timer that runs a script to check the weather using wttr.in every half an hour
```
echo "$(date +%s)=$(curl 'wttr.in/~'$WEATHER_LOC'?format=%l:+%c+%C+%t+(%f),+%w+%m')" > /tmp/weather.txt
```
the `$WEATHER_LOC` environment variable is set in my ~/.profile. I live in a city that has a non-unique name, so I have to set it to `City+State` for wttr.in to get an unambiguous location, or else I get the weather for a city in a different hemisphere.
This creates a file in `/tmp/` that has the date and weather like this
```
 1736566244=Mycity+Mystate: ☁️   Overcast +29°C (+29°C), ↘5km/h 🌔
```
Then I have this as the script that gets called from my i3 config, it reads the weather from /tmp, and sets the `$WTTR` variable to the trimmed weather string only if it is within the last half hour:
```
weather=$(cat /tmp/weather.txt)
age=$(($(date +%s) - ${weather/=*}))
if (( age > 1800 )); then
  export WTTR='"No weather available"'
else
  export WTTR=\""${weather/*:}"\"
fi
rofi -show drun

```
(The escaped quotes were the missing bit of the puzzle). So in my theme I use the environment variable `$WTTR`, and it shows me the last weather report on my drun menu.
```
mainbox {
    children: [ inputbar, textbox-info, listview];
}

textbox-info {
    content: env( WTTR, "No weather available." );  
    expand: false;
}
```
It's working perfectly, thanks for your help. 
![rofi](https://github.com/user-attachments/assets/d698679a-6147-46fc-9472-c8c32a29ecbe)
